src/dbnode/storage/index/aggregate_results.go

```diff
@@ -42,10 +45,56 @@ type aggregatedResults struct {
 	idPool    ident.Pool
 	bytesPool pool.CheckedBytesPool
 
-	pool       AggregateResultsPool
-	valuesPool AggregateValuesPool
-
+	pool             AggregateResultsPool
+	valuesPool       AggregateValuesPool
 	encodedDocReader docs.EncodedDocumentReader
+
+	iOpts   instrument.Options
+	metrics usageMetrics
+}
+
+type usageMetrics struct {
+	total tally.Counter
+
+	totalTerms   tally.Counter
+	dedupedTerms tally.Counter
+
+	totalFields   tally.Counter
+	dedupedFields tally.Counter
+}
+
+type noopCounter struct{}
+
+var _ tally.Counter = (*noopCounter)(nil)
+
+func (*noopCounter) Inc(_ int64) {}
+
+func newUsageMetrics(ns ident.ID, iOpts instrument.Options) usageMetrics {
+	if ns == nil {
```

Comment:
hmm can we have a catch-all metric tag for these instead?

Reply:
I was thinking that, but figured that potentially having a bunch of metrics under `namespace="undefined"` would end up kinda confusing and not add much additional information; can add it if you think it would be worth having though 👍

Reply:
gotcha, when is ns not set? Mostly curious if we want to account for these or not

Reply:
Mostly when we clear out or finalize the results, or if incorrectly initializing when taking an AggregateResults out of the pool, also happened a lot in tests so had to have some sensible defaults for it without updating all the tests.

Reply:
i'd rather not drop metrics.  I don't think an undefined namespace is that confusing. it would allow you to sum across the metrics without namespace, which is probably the common case.